tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -880,3 +880,44 @@ def _query_costs(self):
             r['ResourceId'] = r['ResourceId'].lower()
 
         return result_list
+
+
+class ParentFilter(Filter):
+    """
+    Meta filter that allows you to filter child resources by applying filters to their
+    parent resources.
+
+    You can use any filter supported by corresponding parent resource type.
+
+    :examples:
+
+    Find Azure KeyVault Keys from Key Vaults with ``owner:ProjectA`` tag.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: kv-keys-from-tagged-keyvaults
+            resource: azure.keyvault-keys
+            filters:
+              - type: parent
+                filter:
+                  "tag:owner": ProjectA
+    """
+
+    schema = type_schema(
+        'parent', filter={'type': 'object'}, required=['type'])
+    schema_alias = True
+
+    def __init__(self, data, manager=None):
+        super(ParentFilter, self).__init__(data, manager)
+        self.parent_manager = self.manager.get_parent_manager()
+        self.parent_filter = self.parent_manager.filter_registry.factory(
+            self.data['filter'],
+            self.parent_manager)
+
+    def process(self, resources, event=None):
+        parent_resources = self.parent_filter.process(self.parent_manager.resources())
```

Comment:
it would be much more efficient to use get_resources() with the unique parent ids then fetching all, unless its expected they are already cached.

Reply:
So for non-event based executions we had to fully enumerate the parents already to get the children, so in that case that is OK as long as we add the cache read in to the `def resources` (I know it was missing since I created the resource manager years ago :( ) 

But for event based we'd certainly need some special handling to call `get_resources` with the right ID(s).   If we don't have any children that support events right now we can punt on it (but I think we do?)

Reply:
Good point on events.. I will take a look

Reply:
Created a work item to ensure child resources work properly with event mode. I think today both ARM\non-arm children have issues.

https://github.com/cloud-custodian/cloud-custodian/issues/4617